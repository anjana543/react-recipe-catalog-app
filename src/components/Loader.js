import styled, { keyframes } from 'styled-components';
import PropTypes from 'prop-types';

const rotate360 = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
`;

const Loader = styled.div`
  border: ${(props) => props?.border}px solid #f3f3f3;
  border-radius: 50%;
  border-top: ${(props) => props?.border}px solid
    ${(props) => (props?.color ? props.color : props?.theme?.colors?.primary_600)};
  width: ${(props) => props?.width}px;
  height: ${(props) => props?.height}px;
  animation: ${rotate360} 1000ms linear infinite;
  margin: 0px auto;
`;

Loader.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number,
  border: PropTypes.number,
  color: PropTypes.string,
};

Loader.defaultProps = {
  width: 100,
  height: 100,
  border: 5,
  color: '',
};

export default Loader;
