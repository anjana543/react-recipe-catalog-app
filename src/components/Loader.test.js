import React from 'react';
import { render } from '@testing-library/react';
import Loader from './Loader';

describe('Loader', () => {
  it('the loader works', () => {
    const { container } = render(<Loader />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
