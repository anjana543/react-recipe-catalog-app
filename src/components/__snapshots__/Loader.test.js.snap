// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader it works 1`] = `
<div
  class="sc-bdnylx jKMVEP"
  color=""
  height="100"
  width="100"
/>
`;

exports[`Loader the loader works 1`] = `
<div
  class="sc-bdnylx jKMVEP"
  color=""
  height="100"
  width="100"
/>
`;
